src/treelearner/monotone_constraints.hpp

```diff
@@ -55,14 +92,368 @@ class LeafConstraints {
         entries_[new_leaf].UpdateMin(mid);
       }
     }
+    return std::vector<int>();
   }
 
-  const ConstraintEntry& Get(int leaf_idx) const { return entries_[leaf_idx]; }
+  const ConstraintEntry& Get(int leaf_idx) const override { return entries_[leaf_idx]; }
 
- private:
+ protected:
   int num_leaves_;
   std::vector<ConstraintEntry> entries_;
 };
 
+class FastLeafConstraints : public BasicLeafConstraints {
```

Comment:
`FastLeafConstraints` is my temporary naming, maybe there is a better name?

Reply:
maybe IntermediateLeafConstraints as I called the method "intermediate" in the config?